libraries/core_libs/node/src/node.cpp

```diff
@@ -226,6 +206,39 @@ void FullNode::start() {
         },
         *rpc_thread_pool_);
   }
+  subscription_pool_ = std::make_unique<util::ThreadPool>(1);
+
+  // GasPricer updater
+  final_chain_->block_finalized_.subscribe(
+      [gas_pricer = as_weak(gas_pricer_)](auto const &res) {
+        if (auto _gas_pricer = gas_pricer.lock()) {
```

Comment:
I would stick to the naming conventions so custom variables start with letter and not with undescore